dodiscover/constraint/skeleton.py

```diff
@@ -0,0 +1,470 @@
+import logging
+from collections import defaultdict
+from itertools import combinations
+from typing import Any, Dict, Iterable, List, Set
+
+import networkx as nx
+import numpy as np
+
+from dodiscover.ci import BaseConditionalIndependenceTest
+
+from ..context import Context
+
+logger = logging.getLogger()
+
+ACCEPTED_MARKOVIAN_SKELETON_METHODS = ["complete", "neighbors", "neighbors_path"]
```

Comment:
What about defining this as enums? Then we can avoid the usual issues with string parameters and the options are easier to discover.

Reply:
Added the enumeration. Lmk wyt.

Reply:
Looks good. You can now also directly use it as a type in the code then, i.e. ```skeleton_method: SkeletonMethods``` and do the comparison using the enums ````if self.skeleton_method == SkeletonMethods.COMPLETE```.

Btw. consider using capital letters for the enum values.